apps/rebar/src/rebar3.erl

```diff
@@ -306,7 +306,12 @@ log_level() ->
                 Di when Di == false; Di == "" ->
                     case os:getenv("DEBUG") of
                         D when D == false; D == "" ->
-                            rebar_log:default_level();
+                            try
+                                {ok, L} = application:get_env(rebar, log_level),
+                                 rebar_log:atom_to_level(L)
+                            catch
+                                _:_ -> rebar_log:default_level()
+                            end;
```

Comment:
It looks like that it can be simplify like:
```suggestion
                                 rebar_log:atom_to_level(application:get_env(rebar, log_level, info));
```

Reply:
> It looks like that it can be simplify like:

No, even if you call get_env(rebar, log_level, rebar_log:default_level()) so that this code does not hardcode the info value, someone may still put an invalid value in the configuration, and in that case I also want to substitute the default rather than failing to start.

Reply:
For this case you can change logic in to_level function:
```erlang
atom_to_level(Level) ->
    case Level of
        error -> ?ERROR_LEVEL;
        warn  -> ?WARN_LEVEL;
        debug -> ?DEBUG_LEVEL;
        diagnostic -> ?DIAGNOSTIC_LEVEL;
        info  -> ?INFO_LEVEL;
        _ -> ? INFO_LEVEL
    end.
 ```

Reply:
Yes, but that is also not appropriate. The conversion function should work on defined levels only.

Reply:
But you doing totally same in catch block - if set incorrect value and it will crashed over missing matching - then use default value - default value is `?INFO_LEVEL` based on the code.

Reply:
> But you doing totally same in catch block - if set incorrect value and it will crashed over missing matching - then use default value - default value is `?INFO_LEVEL` based on the code.

Those modules and functions have very different areas of responsibility.